From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: OpenWrt Build System <openwrt@openwrt.org>
Date: Sun, 24 Nov 2024 00:00:00 +0000
Subject: [PATCH] cmake: explicitly link nl80211 and rtnl modules against libnl-tiny

The nl80211 and rtnl modules use symbols from libnl-tiny but were not
being linked against it when the CMake find_library() call failed to
populate the ${libnl_tiny} variable. This resulted in undefined symbol
errors at runtime when loading the modules.

Fix this by explicitly specifying the nl-tiny library name in
target_link_libraries() calls instead of relying on the CMake variable.

Fixes runtime error:
  Unable to dlopen file '/usr/lib/ucode/nl80211.so': Error relocating
  /usr/lib/ucode/nl80211.so: nl_socket_free: symbol not found

Signed-off-by: OpenWrt Build System <openwrt@openwrt.org>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -189,7 +189,7 @@ if(RTNL_SUPPORT)
   add_library(rtnl_lib MODULE lib/rtnl.c)
   set_target_properties(rtnl_lib PROPERTIES OUTPUT_NAME rtnl PREFIX "")
   target_link_options(rtnl_lib PRIVATE ${UCODE_MODULE_LINK_OPTIONS})
-  target_link_libraries(rtnl_lib ${libnl_tiny} ${libubox})
+  target_link_libraries(rtnl_lib nl-tiny ubox)
 endif()

 if(NL80211_SUPPORT)
@@ -199,6 +199,6 @@ if(NL80211_SUPPORT)
   add_library(nl80211_lib MODULE lib/nl80211.c)
   set_target_properties(nl80211_lib PROPERTIES OUTPUT_NAME nl80211 PREFIX "")
   target_link_options(nl80211_lib PRIVATE ${UCODE_MODULE_LINK_OPTIONS})
-  target_link_libraries(nl80211_lib ${libnl_tiny} ${libubox})
+  target_link_libraries(nl80211_lib nl-tiny ubox)
 endif()

